staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -246,7 +259,42 @@ func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) error {
 	return errs[0]
 }
 
+// whether the namespace is exempted from the webhook.
+func (a *GenericAdmissionWebhook) exemptNamespace(h *v1alpha1.ExternalAdmissionHook, attr admission.Attributes) (bool, error) {
+	namespaceName := attr.GetNamespace()
+	if namespaceName == "" {
+		// if the request is root scoped, it is not exempted.
+		return false, nil
+	}
+	selector, err := metav1.LabelSelectorAsSelector(h.NamespaceSelector)
```

Comment:
Seems like you could precompute this to avoid creating it on every request.

Reply:
I'm not sure how to optimize the computation. Maybe using a cache?

Reply:
I added a todo there. We can implement in another pull or get a benchmark first. Other comments are addressed.

Reply:
Yup, no one has config exercising this, so it doesn't need to be fast in this PR. Will add a line in the spreadsheet to cache this. (mostly thinking of garbage)